app/controllers/foreman_tasks/api/tasks_controller.rb

```diff
@@ -141,6 +141,33 @@ def bulk_cancel
         }
       end
 
+      api :POST, '/tasks/bulk_stop', N_('Stop all stoppable tasks')
+      param :search, String, :desc => N_('Stop tasks matching search string')
+      param :task_ids, Array, :desc => N_('Stop specific tasks by ID')
+      def bulk_stop
+        if params[:search].nil? && params[:task_ids].nil?
+          raise BadRequest, _('Please provide at least one of search or task_ids parameters in the request')
+        end
+
+        filtered_scope = bulk_scope
+        skipped, to_stop = filtered_scope.partition { |task| task.state == "stopped" }
+        to_stop.each { |task| task.update({ state: :stopped }) }
```

Comment:
Since we're actually loading the records, we should use something like [find_each](https://api.rubyonrails.org/classes/ActiveRecord/Batches.html#method-i-find_each) so we don't eat too much memory if there's a lot of tasks to be stopped.

Reply:
So should we only send a number instead of all the records at the end?

Reply:
Yes, unless there's a reason why we would want all the details that I'm not seeing right now